org.eclipse.zest.core/src/org/eclipse/zest/core/widgets/GraphConnection.java

```diff
@@ -583,7 +583,10 @@ private void doUpdateFigure(Connection connection) {
 		connectionShape.setLineStyle(getLineStyle());
 
 		if (this.getText() != null || this.getImage() != null) {
```

Comment:
Wouldn't it be more correct to add a this.connectionLabel != null into this if instead of just creating a label that is not shown and not used?

Reply:
i believe the label being null here does say that its not used. Its private so it cannot be initialized by the overridden createFigure method, but could still be used when e.g. the text or the image are set (i.e., the code after the check).

Reply:
I did a second check where connectionLabel is used. And it is only used in the doCreateFigure and here in this method. With your change you are creating a label but never using it. Therefore you are just wasting memory. By simply adding a null check to the if the whole block (which could be moved to an one method). would not be executed. This solves the problem of connectionLabel beeing null and does not waste memory.

Reply:
Ah I see now, you are right.